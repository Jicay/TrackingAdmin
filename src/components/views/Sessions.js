import React, {Component} from "react";
import SessionInfo from "./SessionInfo";

export default class Sessions extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const {getSessions} = this.props;
        getSessions();
    }

    render() {
        const {sessions} = this.props;
        console.log(sessions);
        if (sessions) {
            return (
                <div className="container">
                    {sessions.map(({id, creationDate}) => (
                        <SessionInfo id={id} date={creationDate}/>
                    ))}
                </div>
            );
        } else {
            return (
                <div className="container">
                    Attente de chargement
                </div>
            )
        }
    }
}