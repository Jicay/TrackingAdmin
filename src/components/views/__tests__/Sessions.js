import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router";
import Sessions from "../Sessions";

it('renders correctly', () => {
    const sessions = [{
        id: 45,
        creationDate: '2017-02-12 12:52:23'
    }, {
        id: 85,
        creationDate: '2019-03-29 21:45:10'
    }]

    const tree = renderer
        .create(<MemoryRouter><Sessions getSessions={jest.fn()} sessions={sessions}/></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});