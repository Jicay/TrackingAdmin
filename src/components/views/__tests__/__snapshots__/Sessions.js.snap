// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      45
    </div>
    <div
      className="col"
    >
      2017-02-12 12:52:23
    </div>
    <div
      className="col"
    >
      <a
        href="/sessions/45"
        onClick={[Function]}
      >
        Voir l'état de la session
      </a>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      85
    </div>
    <div
      className="col"
    >
      2019-03-29 21:45:10
    </div>
    <div
      className="col"
    >
      <a
        href="/sessions/85"
        onClick={[Function]}
      >
        Voir l'état de la session
      </a>
    </div>
  </div>
</div>
`;
